tests/unit/test_bruteforce_multi.cpp

```diff
@@ -0,0 +1,1492 @@
+
+    ASSERT_EQ(iteration_num, n_labels / n_res);
+    VecSimBatchIterator_Free(batchIterator);
+    ASSERT_EQ(VecSimIndex_IndexSize(index), n_labels * perLabel);
+    ASSERT_EQ(VecSimIndex_Info(index).bfInfo.indexLabelCount, n_labels);
+
+    VecSimIndex_Free(index);
+}
+
+TEST_F(BruteForceMultiTest, brute_get_distance) {
+    size_t n_labels = 2;
+    size_t dim = 2;
+    size_t numIndex = 3;
+    VecSimIndex *index[numIndex];
+    std::vector<double> distances;
+
+    float v1_0[] = {M_PI, M_PI};
+    float v2_0[] = {M_E, M_E};
+    float v3_1[] = {M_PI, M_E};
+    float v4_1[] = {M_SQRT2, -M_SQRT2};
+
+    VecSimParams params{
+        .algo = VecSimAlgo_BF,
+        .bfParams =
+            BFParams{.type = VecSimType_FLOAT32, .dim = dim, .multi = true, .initialCapacity = 4}};
+
+    for (size_t i = 0; i < numIndex; i++) {
+        params.bfParams.metric = (VecSimMetric)i;
+        index[i] = VecSimIndex_New(&params);
+        VecSimIndex_AddVector(index[i], v1_0, 0);
+        VecSimIndex_AddVector(index[i], v2_0, 0);
+        VecSimIndex_AddVector(index[i], v3_1, 1);
+        VecSimIndex_AddVector(index[i], v4_1, 1);
+        ASSERT_EQ(VecSimIndex_IndexSize(index[i]), 4);
+    }
+
+    void *query = v1_0;
+    void *norm = v2_0;                               // {e, e}
+    VecSim_Normalize(norm, dim, VecSimType_FLOAT32); // now {1/sqrt(2), 1/sqrt(2)}
+    ASSERT_FLOAT_EQ(((float *)norm)[0], 1.0f / sqrt(2.0f));
+    ASSERT_FLOAT_EQ(((float *)norm)[1], 1.0f / sqrt(2.0f));
+    double dist;
+
+    // VecSimMetric_L2
+    // distances are [[0.000, 0.358], [0.179, 23.739]]
+    // minimum of each label are:
+    distances = {0, 0.1791922003030777};
+    for (size_t i = 0; i < n_labels; i++) {
+        dist = VecSimIndex_GetDistanceFrom(index[VecSimMetric_L2], i, query);
+        ASSERT_DOUBLE_EQ(dist, distances[i]);
+    }
+
+    // VecSimMetric_IP
+    // distances are [[-18.739, -16.079], [-17.409, 1.000]]
+    // minimum of each label are:
+    distances = {-18.73921012878418, -17.409339904785156};
+    for (size_t i = 0; i < n_labels; i++) {
+        dist = VecSimIndex_GetDistanceFrom(index[VecSimMetric_IP], i, query);
+        ASSERT_DOUBLE_EQ(dist, distances[i]);
+    }
+
+    // VecSimMetric_Cosine
+    // distances are [[5.960e-08, 5.960e-08], [0.0026, 1.000]]
+    // minimum of each label are:
+    distances = {5.9604644775390625e-08, 0.0025991201400756836};
+    for (size_t i = 0; i < n_labels; i++) {
+        dist = VecSimIndex_GetDistanceFrom(index[VecSimMetric_Cosine], i, norm);
+        ASSERT_DOUBLE_EQ(dist, distances[i]);
+    }
+
+    // Bad values
+    dist = VecSimIndex_GetDistanceFrom(index[VecSimMetric_Cosine], -1, norm);
+    ASSERT_TRUE(std::isnan(dist));
+    dist = VecSimIndex_GetDistanceFrom(index[VecSimMetric_L2], 46, query);
+    ASSERT_TRUE(std::isnan(dist));
+
+    // Clean-up.
+    for (size_t i = 0; i < numIndex; i++) {
+        VecSimIndex_Free(index[i]);
+    }
+}
+
+// TEST_F(BruteForceMultiTest, testCosine) {
```

Comment:
what about this one?

Reply:
I just noticed that myself… I’ll add it in